drivers/usb/device/usb_dc_stm32.c

```diff
@@ -332,11 +351,13 @@ static int usb_dc_stm32_init(void)
 #endif
 	usb_dc_stm32_state.pcd.Init.dev_endpoints = DT_USB_NUM_BIDIR_ENDPOINTS;
 	usb_dc_stm32_state.pcd.Init.speed = usb_dc_stm32_get_maximum_speed();
+#ifndef USB_OTG_FS
```

Comment:
Because [`USB_OTG_CfgTypeDef`](https://github.com/zephyrproject-rtos/hal_stm32/blob/master/stm32cube/stm32f1xx/drivers/include/stm32f1xx_ll_usb.h#L105) (for STM32F1xx with `USB_OTG_FS` defined) don't have a `phy_itface` member... (they do when `USB` is defined)

I might have got something wrong here...

Reply:
For reference:
```
/home/user/zephyrproject/zephyr/drivers/usb/device/usb_dc_stm32.c: In function 'usb_dc_stm32_init':
/home/user/zephyrproject/zephyr/drivers/usb/device/usb_dc_stm32.c:357:29: error: 'PCD_InitTypeDef {aka struct <anonymous>}' has no member named 'phy_itface'
  usb_dc_stm32_state.pcd.Init.phy_itface = PCD_PHY_EMBEDDED;
                             ^
```

Reply:
You are right. I totally missed that. However, it may break the driver for other series (e.g. L4) where `USB_OTG_CfgTypeDef` has a `phy_itface` member.

Reply:
Eeh, I was concerned about that. I'll look at options and get back to you :+1:

Reply:
Variants of the `USB_OTG_CfgTypeDef` structure:
- `.../stm32f1xx/drivers/include/stm32f1xx_ll_usb.h` - `phy_itface` *not present*
- `.../stm32f2xx/drivers/include/stm32f2xx_ll_usb.h` - `phy_itface` present
- `.../stm32f4xx/drivers/include/stm32f4xx_ll_usb.h` - `phy_itface` present
- `.../stm32f7xx/drivers/include/stm32f7xx_ll_usb.h` - `phy_itface` present
- `.../stm32h7xx/drivers/include/stm32h7xx_ll_usb.h` - `phy_itface` present
- `.../stm32l4xx/drivers/include/stm32l4xx_ll_usb.h` - `phy_itface` present

This makes STM32F1xx the outlier...

I think the correct solution here is to swap `#ifndef USB_OTG_FS` for `#ifndef CONFIG_SOC_SERIES_STM32F1X` (as per `soc/arm/st_stm32/stm32f1/Kconfig.series`).

Let me know what you think.

Reply:
It is include after the latest update of stm32 modules. Did you update?